### PR TITLE
Accounts rescan mode scanner

### PR DESCRIPTION
**Pull request by chernetsky**
Closes SCAN-351

### CODE REVIEW

File: apps/crawler/src/scanners/accounts-scanner.service.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    @InjectRepository(Account)
+    private accountsRepository: Repository<Account>,
+    private configService: ConfigService,
+    private sdkService: SdkService,
+  ) {}
+
+  async scan() {
+    this.logger.log('Start full scan...');
+
+    const PARALLEL_TASKS = this.configService.get('PARALLEL_TASKS') | 10;
```
Comment (sswebcoder) [inline apps/crawler/src/scanners/accounts-scanner.service.ts:25]: Typo? Mabe you mean || "logic or" instead of "bit or".

Comment (chernetsky) [reply apps/crawler/src/scanners/accounts-scanner.service.ts]: I felt that strange feeling that something's wrong...) Fixed.

File: apps/crawler/src/scanners/accounts-scanner.service.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            }),
+          ),
+        );
+
+        this.logger.verbose(
+          `Processed ${accountsIdsForProcessing.length} accounts`,
+        );
+      }
+    } catch (err) {
+      this.logger.error(err);
```
Comment (sswebcoder) [inline apps/crawler/src/scanners/accounts-scanner.service.ts:46]: If we not using Sentry as main logger, we should add Sentry logging for all exceptions manually.

File: apps/crawler/src/scanners/accounts-scanner.service.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      this.logger.error({ ...log, error: err.message });
+    }
+  }
+
+  private async getBalancesData(
+    accountId: string,
+  ): Promise<AllBalances | null> {
+    const result = await this.sdkService.getAccountBalances(accountId);
+
+    return result ? result : null;
```
Comment (sswebcoder) [inline apps/crawler/src/scanners/accounts-scanner.service.ts:86]: What kind of value we have if getAccountBalances can't return balances info? Undefined/false?

I mean, why we can't return result as is.

File: apps/crawler/src/scanners/accounts-scanner.service.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      account_id: address,
+      account_id_normalized: normalizeSubstrateAddress(address),
+      available_balance: availableBalance.amount,
+      free_balance: freeBalance.amount,
+      locked_balance: lockedBalance.amount,
+      timestamp: String(timestamp),
+      block_height: String(blockNumber),
+
+      // todo: no data from sdk? Critical?
+      nonce: null,
```
Comment (sswebcoder) [inline apps/crawler/src/scanners/accounts-scanner.service.ts:109]: Nonce is not needed any more.

File: apps/crawler/src/scanners/accounts-scanner.service.ts (hunk 2 above)
Comment (chernetsky) [reply apps/crawler/src/scanners/accounts-scanner.service.ts]: Added. There was no santry at the moment of wrighting this code.

File: apps/crawler/src/skd/sdk.service.ts [hunk 5]
```diff
@@ -8,14 +8,15 @@ import {
   TokenByIdResult,
   TokenPropertiesResult,
 } from '@unique-nft/sdk/tokens';
+import { ConfigService } from '@nestjs/config';
```
Comment (sswebcoder) [inline apps/crawler/src/skd/sdk.service.ts:11]: 'skd' in file path looks like typo.

File: apps/crawler/src/scanners/accounts-scanner.service.ts (hunk 3 above)
Comment (chernetsky) [reply apps/crawler/src/scanners/accounts-scanner.service.ts]: Fixed. No empty value expected.

File: apps/crawler/src/skd/sdk.service.ts (hunk 5 above)
Comment (sswebcoder) [reply apps/crawler/src/skd/sdk.service.ts:11]: image

File: apps/crawler/src/scanners/accounts-scanner.service.ts (hunk 4 above)
Comment (chernetsky) [reply apps/crawler/src/scanners/accounts-scanner.service.ts]: Ok, fixed.

File: apps/crawler/src/skd/sdk.service.ts (hunk 5 above)
Comment (chernetsky) [reply apps/crawler/src/skd/sdk.service.ts:11]: Yopt!

File: apps/crawler/src/subscribers/accounts-subscriber.service.ts [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { InjectRepository } from '@nestjs/typeorm';
 import { Store } from '@subsquid/typeorm-store';
 import { EventHandlerContext } from '@subsquid/substrate-processor';
 import { Account } from '@entities/Account';
-import { SdkService } from '../sdk.service';
+import { SdkService } from '../skd/sdk.service';
```
Comment (sswebcoder) [inline apps/crawler/src/subscribers/accounts-subscriber.service.ts]: skd?

File: apps/crawler/src/subscribers/collections-subscriber.service.ts [hunk 7]
```diff
@@ -17,7 +17,7 @@ import {
   CollectionProperty,
   UniqueCollectionSchemaDecoded,
 } from '@unique-nft/sdk/tokens';
-import { SdkService } from '../sdk.service';
+import { SdkService } from '../skd/sdk.service';
```
Comment (sswebcoder) [inline apps/crawler/src/subscribers/collections-subscriber.service.ts]: skd?

File: apps/crawler/src/skd/sdk.service.ts (hunk 5 above)
Comment (chernetsky) [reply apps/crawler/src/skd/sdk.service.ts]: fixed

File: apps/crawler/src/subscribers/accounts-subscriber.service.ts (hunk 6 above)
Comment (chernetsky) [reply apps/crawler/src/subscribers/accounts-subscriber.service.ts]: fixed

File: apps/crawler/src/subscribers/collections-subscriber.service.ts (hunk 7 above)
Comment (chernetsky) [reply apps/crawler/src/subscribers/collections-subscriber.service.ts]: fixed

**Review by sswebcoder — APPROVED** [review]

**Pull request merged**